content/spin/plugin-authoring.md

```diff
@@ -70,6 +71,24 @@ Spin plugins are implemented as a manifest that points to one or more `.tar.gz`
 1. Create tar archives of the executables for the platforms you want to support
 2. Compose a manifest that describes the plugin and lists the URLs for those tar archives
 
+### Environment Variables Available to the Plugin Executable
+
+When Spin launches a plugin executable, it sets a number of environment variables on the plugin process. These contain information about the Spin binary that the user ran to launch your plugin. Your plugin can use them to detect the Spin environment and adjust its behavior if necessary. The variables set are:
```

Comment:
Regarding the `that the user ran to launch your plugin` are we talking about environment variables that have already been created based on another user's machine i.e. when I launch, my system's environment variables get set behind the scenes (to whatever the creator of the plugin pre-programmed)?

Reply:
Let me explain the context and maybe you can help me improve the wording...

* Suppose there is a plugin called `honk`.
* When the user runs `spin honk`, that runs `spin`, passing the subcommand `honk`.
* Spin recognises that `honk` is not a built-in command, so it checks if it has a plugin called `honk`. Which it does.
* Spin starts the `honk` plugin as a child process.  As part of starting this process, Spin sets a bunch of environment variables on the child process.  These EVs contain information about the current Spin process (the parent).
* The `honk` plugin code runs, with these environment variables set.

The `spin` process in the second bullet, i.e. the one that the user ran when they typed `spin honk`, is what I meant by "the Spin binary that the user ran to launch your plugin."

The EVs come from _this_ Spin process, not from anything programmed by the plugin author.  It's like, as Spin kicks off the plugin, it tells the plugin "hey did you know you can call me back at /home/ivan/github/spin/target/debug/spin? and that my version number is 1.4.0?"

Does that make more sense... or _even less sense_...?